Networking/Networking/Model/MetaData.swift

```diff
@@ -39,18 +47,92 @@ private extension MetaData {
     }
 }
 
-private extension AnyCodable {
-    func getAsString() -> String {
-        switch value {
-        case let string as String:
-            return string
-        default:
-            if JSONSerialization.isValidJSONObject(value) {
-                let data = (try? JSONSerialization.data(withJSONObject: value)) ?? Data()
-                return String(data: data, encoding: .utf8) ?? ""
-            } else {
-                return "\(value)"
+public struct MetaDataValue: Codable, Equatable, Sendable {
```

Comment:
We expose a `struct` instead of exposing the enum directly just to make sure we have some validation, and we no consumer can bypass it by instantiating the enum directly, ie: `.json("invalid json")`, this is actually because as far as I found, Swift doesn't allow to have private initializer for the enum cases (unlike Kotlin with its `sealed interface` feature).

Reply:
I think this is a good architectural choice in terms of replicating `sealed interface` in Kotlin.

With that being said, I'm curious if Swift has a commonly used alternative for this purpose. cc @selanthiraiyan for opinion.

Reply:
> We expose a struct instead of exposing the enum directly just to make sure we have some validation, and we no consumer can bypass it by instantiating the enum directly, ie: .json("invalid json")

I hope you are referring to the `ValueHolder` enum here. I don't know of any way to prevent directly initialising an enum with an associated value. 

The `MetaDataValue` wrapper technique looks good to me.